packages/noya-state/src/reducers/application.ts

```diff
@@ -1099,26 +782,92 @@ export function reducer(
 
         sharedSwatches.forEach((swatch: Sketch.Swatch) => {
           if (ids.includes(swatch.do_objectID)) {
-            swatch.name = name;
+            const newValue =
+              mode === 'replace' ? alpha : swatch.value.alpha + alpha;
+
+            swatch.value.alpha = Math.min(Math.max(0, newValue), 1);
           }
         });
       });
     }
-    case 'setSwatchOpacity': {
-      const [, id, alpha, mode = 'replace'] = action;
+    case 'addLayerStyle': {
+      return produce(state, (state) => {
+        const layerStyles = state.sketch.document.layerStyles ?? {
+          _class: 'sharedStyleContainer',
+          do_objectID: uuid(),
+          objects: [],
+        };
 
-      const ids = typeof id === 'string' ? [id] : id;
+        const style = {} as Sketch.Style;
+        const fill = {} as Sketch.Fill;
+        const contextSettings = {} as Sketch.GraphicsContextSettings;
```

Comment:
I added a need style model instead. "style.json" 
![image](https://user-images.githubusercontent.com/16924758/113464133-1cd3aa80-93f0-11eb-93c1-4e5e69350165.png)

Reply:
![image](https://user-images.githubusercontent.com/16924758/113464172-53112a00-93f0-11eb-93b3-f1664bf2ee49.png)
Sorry just change it to this so it had an different id

Reply:
I tried a way of changing the id of the style but it gives an error about being read only.

Reply:
This looks good! I'm not sure about the readonly error, but maybe it will work if you use `produce`?

```ts
const sharedStyle: Sketch.SharedStyle = {
  _class: 'sharedStyle',
  do_objectID: uuid(),
  name: 'New Style Layer',
  value: produce(Models.style, style => {
    style.do_objectID = uuid()
  }),
};
```

Reply:
Oh! Now I know that ! 💯